pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -201,16 +237,86 @@ func (d *PmemCSIDriver) reconcileDeploymentChanges(r *ReconcileDeployment, chang
 	return false, nil
 }
 
-func (d *PmemCSIDriver) deployObjects(r *ReconcileDeployment) error {
-	objects, err := d.getDeploymentObjects()
-	if err != nil {
-		return err
+func (d *PmemCSIDriver) deleteObsoleteObjects(r *ReconcileDeployment, newObjects []apiruntime.Object) error {
+	type NameGVK struct {
+		name      string
+		namespace string
+		gvk       schema.GroupVersionKind
 	}
-	for _, obj := range objects {
-		if err := r.Create(obj); err != nil {
+
+	foundList := []NameGVK{}
+	obsoleteList := []NameGVK{}
+
+	for gvk, clusterScoped := range allObjectTypes {
+		list := &unstructured.UnstructuredList{}
+		list.SetGroupVersionKind(gvk)
+		opts := client.ListOptions{}
+		if !clusterScoped {
+			opts.Namespace = d.namespace
+		}
```

Comment:
I don't think we need the `clusterScoped` boolean and thus can make our type list simpler. My expectation is that `opts.Namespace` will be ignored when it isn't needed.